### PR TITLE
Tkinter Workshop

### PR DESCRIPTION
**Pull request by bezlin6mechminerz**
Hackclub :- sacred hearts computer club

### CODE REVIEW

**Comment by faisalsayed10** [comment]
Hi @bezlin6mechminerz! by the way, are you a member or a leader of your club?

**Comment by bezlin6mechminerz** [comment]
Leader

**Comment by bezlin6mechminerz** [comment]
Hackclub is for real @zachlatta visited our school

**Comment by sampoder** [comment]
Not all of the requested changes were made, please only reopen the PR when all requested changes were made.

**Comment by sampoder** [comment]
@lachlanjc can you close this? @bezlin6mechminerz refer to my comment on #1316 and make sure all changes are made.

**Comment by bezlin6mechminerz** [comment]
wait!

**Comment by bezlin6mechminerz** [comment]
I have been spending all this time and fixed the grammar and phrases I will be updating it soon

**Comment by bezlin6mechminerz** [comment]
I have made all the changes and i think its almost ready.

File: workshops/Tkinter Workshop/README.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+We have passed the random quote to this function
+
+```python
+text=quote
+```
+
+Here we gave text as the random quote passed from the button
+
+If you don't understand it, just go to the live demo and see the button workflow [here](https://repl.it/@bezlin/graphics#main.py).
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:184]:
```suggestion
If you don't understand, just go to the live demo and see the button workflow [here](https://repl.it/@bezlin/graphics#main.py).
```

File: workshops/Tkinter Workshop/README.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+```
+
+You can specify the name of the window inside the function in quotes. Now, let's give the window color. If you like white which is the default ,go with it or do this.
+
+```python
+window.config(bg="black")
+```
+
+I gave black. You can also specify the color in hex code. RGB too.
+window.config means we are configuring our window and setting up its bg which is short for background. Now let's display the Quotes Hub in the window we made. For that we use the Label function.
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:94]:
```suggestion
I used black, but you can also specify the color as a hex code or RGB value.
`window.config` means we are configuring our window and setting up its bg which is short for background. Now let's display the Quotes Hub in the window we made. For that we use the Label function.
```

File: workshops/Tkinter Workshop/README.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+```python
+tk.Label(window, font=("Helvetica", 60, "bold"),text="QUOTES HUB", bg="black", fg="white").pack()
+```
+
+The Label function is used to display text on the window. The Label function takes parameters such as font,text,background color(bg) ,font color(fg) .The first parameter you should give is the name of the window in which you need this text to be displayed.
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:100]: Spaces missing/mistakes here

File: workshops/Tkinter Workshop/README.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+font=("Helvetica", 60, "bold")
+```
+Here, in this piece of code, we are setting up the style of the text. font=("font name", font-size, font-weight).
+Now, lets make a button.
+
+```python
+tk.Button(window,command=lambda:fun1(random.choice(quotelist)),text="Open",bg="aqua",fg="black",width=20).place(x=315,y=300)
+```
+
+The Tkinter library consists of A Button function which is used to display a button on the window. The button function  takes many parameters such as text,bg,fg,width,height etc... The first parameter you should give is the window in which you need this button to be displayed. Place() function is used to place the element in the particular coordinate we need it to be ,for example:  button.place(x,y). The main thing here is the command parameter inside the button function
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:114]: Spaces missing/mistakes here

File: workshops/Tkinter Workshop/README.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+window.minsize(800, 560)
+window.title("Quotes Hub")
+window.config(bg="black")
+tk.Label(window, font=("Helvetica", 60, "bold"),
+         text="QUOTES HUB", bg="black", fg="white").pack()
+tk.Button(window,command=lambda:clicked(random.choice(quotelist)),text="Open",bg="aqua",fg="black",width=20).place(x=315,y=300)
+window.mainloop()
+```
+
+In the output, an error will be shown,that the function clicked is not declared. So let us make the button work. You need to put this code above the window declaration ,only then will it work.
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:157]: Awkward grammar/punctuation mistakes here

File: workshops/Tkinter Workshop/README.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+window.title("Quotes Hub")
+window.config(bg="black")
+tk.Label(window, font=("Helvetica", 60, "bold"),
+         text="QUOTES HUB", bg="black", fg="white").pack()
+tk.Button(window,command=lambda:clicked(random.choice(quotelist)),text="Open",bg="aqua",fg="black",width=20).place(x=315,y=300)
+window.mainloop()
+
+```
+
+Its simple, you just have to go through the functions and workflow.
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:221]:
```suggestion
It’s simple, you just have to go through the functions and workflow.
```

File: workshops/Tkinter Workshop/README.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+![ezgif com-gif-to-mp4](https://cloud-5m2nwfs8r.vercel.app/ezgif.com-video-to-gif-4.gif)
+
+## Hacking time!
+
+Now you know Tkinter a bit. You should not stop here. You need to learn more and to master in UIs you need to practice a lot. The best way to learn it is to look at the world, think of an idea and then try to make it.
+
+Lastly, don't get give up.
```
Comment (lachlanjc) [inline workshops/Tkinter Workshop/README.md:229]: Grammar

**Review by lachlanjc — CHANGES_REQUESTED** [review]
Thanks for the contribution, Bezlin! While I really like the idea of this workshop, the phrasing, grammar, & punctuation need some significant editing. I left a few comments here but there’s a lot more. Maybe you could go through this workshop with a friend for editing? Since perfect grammar/punctuation are a requirement for our workshop bounties, I unfortunately am closing this for now, but if you can get that fixed up we’d love to have your workshop.

**Comment by bezlin6mechminerz** [comment]
OK

**Comment by bezlin6mechminerz** [comment]
I will make changes and come back again

**Comment by sampoder** [comment]
thanks Bezlin!

**Pull request closed without merging**